task4/src/main/java/BarcodeAnalyzer.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Class of analyzing barcodes EAN-13 and EAN-8.
+ * Input barcode: text parameter,
+ * Barcode type - numeric parameter: 1 of EAN-8, EAN-2 to 13.
+ * The program is note that: some barcode scanners can cut in a bar code
+ * first leading zero (eg. Instead of the code 0075678164125 send 075678164125 ).
+ * Some products (eg. Newspapers) have additional codes (ie. Add-on'y), the scanner appends them directly to the appropriate barcode
+ * (Read add-on "12" and the appropriate code "6920702707721" is sent as "692070270772112").
+ * Add-on'y may occur for both EAN-8 and EAN I-13.
+ * Any errors are signaled with exceptions.
+ * For information on barcodes: http://pl.wikipedia.org/wiki/EAN
+ *
+ * @author KamilBest
+ */
+class BarcodeAnalyzer {
+    protected BarcodeScannerInput input;
+    protected String barcode;
+    protected int barcodeType;
+    private char[] barcodeToAnalyze;
+
+    /**
+     * Constructor for taking input from user
+     *
+     * @param BarcodeScannerInput input
+     */
+    public BarcodeAnalyzer(BarcodeScannerInput input){
+        this.input=input;
+        this.barcode=input.getBarcode();
```

Comment:
Wzorowałem się na kodzie, który mi pokazywałeś :
public FizzBuzz(Output output) {
        this.output = output;
        this.from = 1;
        this.to= 100;
    }

A gdzie powinienem pobierać dane?, BarcodeAnalyzer to główna klasa.